e2e/e2e_test.go

```diff
@@ -15,34 +15,134 @@ package e2e
+
+	parcaServerPod, err := kubeClient.CoreV1().Pods("parca").List(ctx, metav1.ListOptions{LabelSelector: labelSelectorParcaServer})
+	if err != nil {
+		return "", "", fmt.Errorf("Unable to fetch pods in parca namespace: %w", err)
+	}
+
+	parcaAgentPod, err := kubeClient.CoreV1().Pods("parca").List(ctx, metav1.ListOptions{LabelSelector: labelSelectorParcaAgent})
+	if err != nil {
+		return "", "", fmt.Errorf("Unable to fetch pods in parca namespace: %w", err)
+	}
+
+	if len(parcaServerPod.Items) == 0 {
+		fmt.Printf("Parca Server Pod not found")
+		return "", "", nil
+	}
+
+	if len(parcaAgentPod.Items) == 0 {
+		fmt.Printf("Parca Agent Pod not found")
+		return "", "", nil
+	}
+
+	return parcaServerPod.Items[0].Name, parcaAgentPod.Items[0].Name, nil
+}
+
+// TODO(sylfrena): Cleanup logs once e2e tests are stabilized
+// TODO(sylfrena): Reduce context timeouts
+// TODO(sylfrena): Use exponential backoff instead.
+func TestGRPCIntegration(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+
+	cfg, err := GetKubeConfig(*kubeconfig)
+	require.NoError(t, err)
+
+	kubeClient, err := kubernetes.NewForConfig(cfg)
+	require.NoError(t, err)
+
+	parcaServer, parcaAgent, err := CheckPodsExist(ctx, kubeClient)
+	if err != nil {
+		t.Logf("pod discovery error: %s", err)
+		require.NoError(t, err)
+	}
+	t.Log("Pods discovered: ", parcaServer, parcaAgent)
+
+	ns := "parca"
+
+	serverCloser, err := StartPortForward(ctx, cfg, "https", parcaServer, ns, "7070")
+	if err != nil {
+		t.Logf("failed to start port forwarding Parca Server: %v", err)
+		require.NoError(t, err)
+	}
+	defer serverCloser()
+
+	agentCloser, err := StartPortForward(ctx, cfg, "https", parcaAgent, ns, "7071")
+	if err != nil {
+		t.Logf("failed to start port forwarding Parca Agent: %v", err)
+		require.NoError(t, err)
+	}
+	defer agentCloser()
+
+	t.Log("Starting tests")
 	conn, err := grpc.Dial("127.0.0.1:7070", grpc.WithTransportCredentials(insecure.NewCredentials()))
 	require.NoError(t, err)
 	defer conn.Close()
 
+	t.Log("Creating query service client")
 	c := pb.NewQueryServiceClient(conn)
-	ctx := context.Background()
 
-	queryRequest := &pb.QueryRangeRequest{
-		Query: `parca_agent_cpu:samples:count:cpu:nanoseconds:delta{container="parca-agent"}`,
+	t.Log("Performing Query Range Request")
+	queryRequestAgent := &pb.QueryRangeRequest{
+		Query: `parca_agent_cpu:samples:count:cpu:nanoseconds:delta`,
 		Start: timestamppb.New(timestamp.Time(0)),
-		End:   timestamppb.New(timestamp.Time(9223372036854775807)),
+		End:   timestamppb.New(timestamp.Time(math.MaxInt64)),
 		Limit: 10,
 	}
 
-	resp, err := c.QueryRange(ctx, queryRequest)
+	for i := 0; i < 10; i++ {
+		ctx, cancel = context.WithTimeout(context.Background(), 1*time.Minute)
+		defer cancel()
+		resp, err := c.QueryRange(ctx, queryRequestAgent)
+		if err != nil {
+			status, ok := status.FromError(err)
```

Comment:
Is this the code that gets returned when the request succeeds but not data was found after all? 
Just want to make sure that something like this isn't missing:
```suggestion
			status, ok := status.FromError(err)
			if ok && status.Code() == codes.Ok && (resp.Series) == 0 {
				t.Log("query range resource not found, retrying in a minute\n", err)
				time.Sleep(time.Minute)
				continue
			}
```

Reply:
Just so I understand the api better, when would such a case be possible? I thought we were already returning a NotFound for empty series here 
https://github.com/parca-dev/parca/blob/e4f5d43956ebfa1d09ebb996c27d65a26734df59/pkg/parcacol/querier.go#L286